2d_solver/msh_parser.py

```diff
@@ -0,0 +1,300 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import numpy as np
+from matplotlib import pyplot as plt
+import matplotlib.lines as lines
+import gmshtranslator.gmshtranslator as gmsh
+
+import lagrange
+import msh_parser
+import isoparam
+import utils
+
+plt.rcParams['figure.figsize']     = 12, 7.5
+plt.rcParams['lines.linewidth']    = 1.5
+plt.rcParams['font.family']        = 'serif'
+plt.rcParams['font.weight']        = 'bold'
+plt.rcParams['font.size']          = 20  
+plt.rcParams['font.sans-serif']    = 'serif'
+plt.rcParams['text.usetex']        = True
+plt.rcParams['axes.linewidth']     = 1.5
+plt.rcParams['axes.titlesize']     = 'medium'
+plt.rcParams['axes.labelsize']     = 'medium'
+
+plt.rcParams['xtick.major.size']   = 8
+plt.rcParams['xtick.minor.size']   = 4
+plt.rcParams['xtick.major.pad']    = 8
+plt.rcParams['xtick.minor.pad']    = 8
+plt.rcParams['xtick.color']        = 'k'
+plt.rcParams['xtick.labelsize']    = 'medium'
+plt.rcParams['xtick.direction']    = 'in'    
+
+plt.rcParams['ytick.major.size']   = 8
+plt.rcParams['ytick.minor.size']   = 4
+plt.rcParams['ytick.major.pad']    = 8
+plt.rcParams['ytick.minor.pad']    = 8
+plt.rcParams['ytick.color']        = 'k'
+plt.rcParams['ytick.labelsize']    = 'medium'
+plt.rcParams['ytick.direction']    = 'in'
+plt.rcParams['text.usetex']        = True
+plt.rcParams['text.latex.unicode'] = True
+
+def read_order_2_msh(msh_file):
+    '''
+    Parses the :math:`2^{nd}` order **.msh** files.
+    
+    Parameters
+    ----------
+    
+    msh_file : str
+               **.msh** file to be parsed
+    
+    Returns
+    -------
+    
+    tuple(np.ndarray, np.ndarray)
+             Tuple of Nodes and Elements respectively.
+             Nodes is a :math:`N \\times 2` array, where, :math:`N` is the
+             total number of Nodes in the mesh. Each node contains it's
+             :math:`(x, y)` coordinates.
+             Elements is a :math:`N_e \\times 8` array which contains the tags
+             of all the nodes which defines each element. A tag of a node is
+             the array index of a node.
+    '''
+    
+    msh_handler = gmsh.gmshTranslator(msh_file)
+    
+    nodes    = []
+    elements = []
+    
+    def is_node(tag, x, y, z, physgroups):
+        return True
+    
+    def save_node (tag, x, y, z):
+        nodes.append([x, y])
+        return
+
+    
+    def is_9_node_quadrangle (eletag, eletype,
+                              physgrp, nodes):
+        return eletype == msh_handler.quadrangle_9_node
+
+    def save_element (eletag, eletype,
+                      physgrp, node_tags):
+        
```

Comment:
Documentation is missing

Reply:
Name of the function is already descriptive.